lib/babe/state.go

```diff
@@ -28,6 +28,7 @@ type BlockState interface {
 	GetHeader(common.Hash) (*types.Header, error)
 	GetBlockByNumber(blockNumber uint) (*types.Block, error)
 	GetBlockByHash(common.Hash) (*types.Block, error)
+	GetBlockHashesBySlot(uint64) ([]common.Hash, error)
```

Comment:
nit adding named arguments and return values is nice for documentation

```suggestion
	GetBlockHashesBySlot(slot uint64) (blockHashes []common.Hash, err error)
```